src/main/scala/com/gd/twitteranalytics/TwitterReportingJob.scala

```diff
@@ -0,0 +1,55 @@
+package com.gd.twitteranalytics
+
+import com.gd.twitteranalytics.util.AppConfigReader
+import org.apache.hadoop.fs.{FileSystem, Path}
+import org.apache.log4j.Logger
+import org.apache.spark.sql.SparkSession
+
+object TwitterReportingJob {
+
+  val log = Logger.getLogger(TwitterReportingJob.getClass.getName)
+
+  def main(args: Array[String]): Unit = {
+
+    val today = java.time.LocalDate.now
+    val dataPath = AppConfigReader.getAppConfigurables(0) + "/event_date=" + today
+    val spark = setSparkSession
+
+    val fs = FileSystem.get(spark.sparkContext.hadoopConfiguration)
+    if(fs.exists(new Path(dataPath))) {
+      val reportSavePath = "twitter-analytics/reporting"
+
+      log.debug("=======> Building Twitter KPI Report for date " + today)
+
+      val tweetsDataFrame = JsonTranslator.getParsedJsonAsDataFrame(spark, dataPath)
+
+      log.debug("=======> Stage 1: Twitter Data Read Complete !")
+      val filteredStatusDf = JsonTranslator.getFilteredDf(spark, tweetsDataFrame)
+
+      val reportFromSql = TwitterReportGenerator.getReportWithSqlProcessing(spark, filteredStatusDf)
+      val reportFromDf = TwitterReportGenerator.getReportWithDataFrameProcessing(spark, filteredStatusDf)
+      val reportFromDs = TwitterReportGenerator.getReportWithDataSetProcessing(spark, filteredStatusDf)
+
+      log.debug("=======> Stage 5: Twitter Report Creation Complete!")
+
+      log.debug("=======> Stage 6: Saving Report on HDFS!")
+      TwitterReportWriter.saveReportToHdfs(reportFromSql, reportSavePath + "/sqlReport")
+      TwitterReportWriter.saveReportToHdfs(reportFromDf, reportSavePath + "/dataFrameReport")
+      TwitterReportWriter.saveReportToHdfs(reportFromDs, reportSavePath + "/dataSetReport")
```

Comment:
no need to run every implementation in the single driver.
please design the code in a way that allows to run the driver with a particular implementation.

Reply:
Are you suggesting every implementation goes through its own driver ?

Reply:
it's up to you how you decide to implement it.
the main point is - current approach has a bit of overhead.